pkg/machine/wsl/machine.go

```diff
@@ -1442,3 +1442,57 @@ func (v *MachineVM) getResources() (resources machine.ResourceConfig) {
 	resources.DiskSize = getDiskSize(v)
 	return
 }
+
+// RemoveAndCleanMachines removes all machine and cleans up any other files associatied with podman machine
+func (p *Provider) RemoveAndCleanMachines() error {
```

Comment:
Why is this function logrus.Error rather then returning it? Looks like it never returns an error?

Reply:
The function should not error just because one machine failed to be removed, but rather keep going to check and remove more machines. I can either logrus.err it or return a list of errors and also error, but I think logrus is cleaner in this caler. I think like 1458 should be a return err though, Ill fix that.

Reply:
If the function never returns an error, then it should not have error in its core.

I think you should do
```
var prevError Error
...
if err != nil {
        if prevError != nil {
                logrus.Errorf(prevErr)
         }
        prevError = err
}
...
Return prevError
```

Reply:
So the calling function would also just logrus the returned error too? 

I updated the function to return an error if it's unable to list machines to remove, so the function will return an error if it fails there, do you think it's still necessary to do what you said?

Reply:
It would logrus all errors except for the last error.  So if only one error, then no logging.  If two errors logus.Error First Error, return second.